src/text-editor-presenter.coffee

```diff
@@ -595,33 +641,24 @@ class TextEditorPresenter
       isVisible = isVisible and @showLineNumbers
     isVisible
 
-  updateLineNumbersState: (tileState, startRow, endRow) ->
-    tileState.lineNumbers ?= {}
-    visibleLineNumberIds = {}
+  isSoftWrappedRow: (bufferRow, screenRow) ->
+    return false if screenRow is 0
 
-    if startRow > 0
-      rowBeforeStartRow = startRow - 1
-      lastBufferRow = @model.bufferRowForScreenRow(rowBeforeStartRow)
-    else
-      lastBufferRow = null
+    @model.bufferRowForScreenRow(screenRow - 1) is bufferRow
```

Comment:
I went ahead and implemented this in 116f92d, as it seemed quite straightforward. Please, let me know if you think this could have any unintended side effects.